src/assets/scripts/airport/Route/RouteModel.js

```diff
@@ -0,0 +1,134 @@
+ * @property SEGMENT_SEPERATION_SYMBOL
+ * @type {string}
+ * @final
+ */
+const SEGMENT_SEPERATION_SYMBOL = '.';
+
+/**
+ * @class RouteModel
+ */
+export default class RouteModel {
+    /**
+     * 'BETHL.GRNPA1.KLAS'
+     *
+     * @for RouteModel
+     * @constructor
+     * @param routeString {string}
+     */
+    constructor(routeString) {
+        if (typeof routeString === 'undefined' || typeof routeString !== 'string') {
+            return;
+        }
+
+        if (!this._isValidRouteString(routeString)) {
+            // eslint-disable-next-line max-len
+            throw new TypeError(`Invalid routeString passed to RouteModel. Expected a routeString of the shape ORIGIN.BASE.DESTINATION but instead received ${routeString}`);
+        }
+
+        /**
+         * @property
+         * @type {string}
+         */
+        this._id = _uniqueId();
+
+        /**
+         * @property origin
+         * @type {string}
+         * @default ''
+         */
+        this.origin = '';
+
+        /**
+         * @property base
+         * @type {string}
+         * @default ''
+         */
+        this.base = '';
+
+        /**
+         * @property destination
+         * @type {string}
+         * @default ''
+         */
+        this.destination = '';
+
+        return this._init(routeString);
+    }
+
+    /**
+     * A single string that represents the entire route
+     *
+     * @property routeString
+     * @return {string}
+     */
+    get routeString() {
+        return `${this.origin}.${this.base}.${this.destination}`;
+    }
+
+    /**
+     * Lifecycle method. Should be run only once on instantiation
+     *
+     * @for RouteModel
+     * @method _init
+     * @param routeString {string}
+     * @private
+     */
+    _init(routeString) {
+        const { origin, base, destination } = this._extractSegmentNamesFromRouteString(routeString);
+
+        this.origin = origin;
+        this.base = base;
+        this.destination = destination;
+
+        return this;
+    }
+
+    /**
+     * Destroy this instance
+     *
+     * @for RouteModel
+     * @method destroy
+     */
+    destroy() {
+        this._id = '';
+        this.origin = '';
+        this.base = '';
+        this.destination = '';
+    }
+
+    /**
+     *
+     * @for RouteModel
+     * @method _extractSegmentNamesFromRouteString
+     * @param routeString {string}
+     * @return {object}
+     * @private
+     */
+    _extractSegmentNamesFromRouteString(routeString) {
+        const routeSegments = routeString.split(SEGMENT_SEPERATION_SYMBOL);
+
+        return {
+            origin: routeSegments[0],
+            base: routeSegments[1],
+            destination: routeSegments[2]
+        };
+    }
+
+    /**
+     *
+     * @for RouteModel
+     * @method _isValidRouteString
+     * @param routeString {string}
+     * @return {boolean}
+     * @private
+     */
+    _isValidRouteString(routeString) {
+        // TODO: replace magic number with enumeration
```

Comment:
(presumably) has to do with the assumptions of the format of the route string. A single leg MUST have either `firstfix..secondfix` OR `firstfix.procedure.secondfix`. The former is handled somewhere else I guess, and the latter must have EXACTLY three elements when `.split(".")` is run on it. Otherwise, the route wouldn't make sense. So perhaps:
`const expected_elements_in_route_string = 3;`
or something more sensibly named (I'm getting sleepy and it's the best I can do)